rigex-front-end/src/components/WellModal/WellModal.js

```diff
@@ -0,0 +1,201 @@
+import React, { useState, useEffect } from 'react';
+import { useMutation } from '@apollo/client';
+import { useDispatch } from 'react-redux';
+import Dialog from '@material-ui/core/Dialog';
+import CloseIcon from '@material-ui/icons/Close';
+
+import { wellActions } from '../../store/wells/action';
+import { UPSERT_WELL } from '../../constants/serviceAPI';
+import './wellModalStyles.scss';
+
+export default function WellModal({
+  wellModalOpen: {
+    type,
+    isOpen,
+    currentName,
+    currentLatitude,
+    currentLongitude,
+    id,
+  },
+  handleWellModalClose,
+  rigId,
+}) {
+  const [wellInfo, setWellInfo] = useState({
+    name: '',
+    latitude: '',
+    longitude: '',
+  });
+  const [isValidInput, setIsValidInput] = useState({
+    name: false,
+    latitude: false,
+    longitude: false,
+  });
+  const [isCreateType, setIsCreateType] = useState(true);
+  const [isActiveSubmit, setIsActiveSubmit] = useState(false);
+  const [upsertWell, { loading, error, data }] = useMutation(UPSERT_WELL);
+  const dispatch = useDispatch();
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const wellInput = {
+      name: wellInfo.name,
+      longitude: parseFloat(wellInfo.longitude),
+      latitude: parseFloat(wellInfo.latitude),
+      rigId: rigId.toString(),
+    };
+
+    if (id) {
+      wellInput.id = id;
+    }
```

Comment:
and what if id doesn't exist? 

inside your const on line 40 add `input: id ? id : {whatever happens if id doesn't exist}`